source/fuzz/transformation_add_type_float.cpp

```diff
@@ -36,6 +36,15 @@ bool TransformationAddTypeFloat::IsApplicable(
     return false;
   }
 
+  // If the float type width is 16 or 64, then the corresponding Float16 or
+  // Float64 capability must be present.
+  if ((message_.width() == 16 &&
+       !ir_context->get_feature_mgr()->HasCapability(SpvCapabilityFloat16)) ||
+      (message_.width() == 64 &&
+       !ir_context->get_feature_mgr()->HasCapability(SpvCapabilityFloat64))) {
+    return false;
+  }
```

Comment:
This is really nice. Sorry to not have suggested this sooner, but could you do a switch statement on the width, include cases for all expected widths (16, 32, 64, (and 8 for int)) where the 32 case just has a break with a comment saying something like `// No capabilities needed.` And then in the default case do:

```
assert(false && "Unexpected float type width");
return false;
```

Or something like that. And do the same for TransformationAddTypeInt.

Reply:
Done.